src/test/helpers/HttpServer.js

```diff
@@ -5,21 +5,53 @@ function HttpServer(config) {
   this.address = config.address || "localhost";
   this.options = {
     data: null,
+    headers: {"Content-Type": "application/json"},
+    returnPayload: false,
     resCode: 200
   };
 }
 
 HttpServer.prototype.start = function (fn) {
-  this.server = http.createServer(function (req, res) {
-    res.writeHead(this.options.resCode, {"Content-Type": "application/json"});
-    res.end(JSON.stringify(this.options.data));
-  }.bind(this)).listen(this.port, this.address, fn);
+
+  this.server = http.createServer();
+
+  this.server.on("request", (req, res) => {
+
+    if (this.options.returnPayload === true) {
+      this.options.data = {
+        payload: "",
+        method: req.method
+      };
+
+      req.addListener("data", chunk => {
+        this.options.data.payload += chunk;
+      });
+
+      req.addListener("end", chunk => {
+        if (chunk) {
+          this.options.data.payload += chunk;
+        }
+        res.writeHead(this.options.resCode, this.options.headers);
+        res.end(JSON.stringify(this.options.data));
+      });
+    } else {
```

Comment:
Unfortunately `req.addListener("end", ...)` doesn't get triggered in the simple scenarion, hence the ugly `else`.